LanguageFeatures/Patterns/shared_case_scope_A01_t02.dart

```diff
@@ -79,6 +81,8 @@ main() {
       print(a);
       break;
     case int a when a == 42:
+//           ^
+// [cfe] unspecified
```

Comment:
I do not understand this error: The log seems to imply that it's a 'Variable pattern 'a' doesn't have the same type in all cases' problem, but I think they should both (line 80 `a` and line 82 `a`) have declared type `int`.

Reply:
It's how CFE behaves now. It reports an additional error here. If you believe that it sholdn't I can revert change in this file. Please confirm

Reply:
Oh, wait, I misread the situation: We have shared-body cases in lines 83, 86, 87 (types `int`, `String`, `int`), so I'd expect an error at line 88 when `a` is used in the body.

But I still don't understand why there would be an error at line 83. It is _not_ an error for multiple cases with a shared body to declare a variable with the same name `a` and different types/finality, as long as it isn't used in the body. If it is used then there will be an error _at the usage_, not in one or more cases.

Reply:
I think we should maintain that there is no error in line 83.

Reply:
Reverted changes in shared_case_scope_A01_t02.dart

Reply:
Thx!